File: src/net/Defracted/AdminUtils/commands/Ban.java
package net.Defracted.AdminUtils.commands;

import net.Defracted.AdminUtils.Main;
import net.Defracted.AdminUtils.util.Formatters;

import org.bukkit.BanList;
import org.bukkit.Bukkit;
import org.bukkit.command.Command;
import org.bukkit.command.CommandExecutor;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import java.util.Date;

// Нужно завершить
public class Ban implements CommandExecutor {
    private Main plugin;

    public Ban(Main instance) {
        this.plugin = instance;
    }

    public boolean onCommand(CommandSender sender, Command cmd, String label, String[] args) {
        if (cmd.getName().equalsIgnoreCase("ban")) {
            if (args.length <= 0) {
                sender.sendMessage(Formatters.chat("&cВы не указали ник!"));
                return true;
            }

            // Получаем цель бана
            String target = args[0];

            // Проверяем наличие длительности
            if (args.length < 2) {
                sender.sendMessage(Formatters.chat("&cВы не указали длительность бана."));
                return true;
            }

            // Получаем срок бана (в минутах)
            int punishmentDuration;
            if (Formatters.isStrNum(args[1])) {
                punishmentDuration = Integer.parseInt(args[1]);
            } else {
                sender.sendMessage(Formatters.chat("&cУкажите срок действия бана в минутах!"));
                return true;
            }

            // Создаём дату, основываясь на указанном времени
            Date punishmentDateExpiration;
            if (punishmentDuration > 0) {
                punishmentDateExpiration = new Date(System.currentTimeMillis() + (punishmentDuration * 60000));
            } else {
                punishmentDateExpiration = null;
            }

            // Получение причины из аргументов
            String reason = "";
            if (args.length <= 2) {
                reason += "не указана.";
            } else {
                for (int i = 2; i < args.length; i++)
                    reason += args[i] + (i != (args.length - 1) ? " " : "");
            }


            String infoExpiration;
            if (punishmentDuration <= 0) {
                infoExpiration = "&fнавсегда";
            } else {
                infoExpiration = "&fдо " + punishmentDateExpiration.toString();
            }

            // Кикаем цель, но если игрок в онлайне
            Player playerTarget = Bukkit.getPlayerExact(target);

            // Баним цель на указанный срок
            Bukkit.getBanList(BanList.Type.NAME).addBan(target, reason, punishmentDateExpiration, sender.getName());

            // Кикаем игрока, если он в онлайне
            if (playerTarget != null)
                playerTarget.kickPlayer(Formatters.chat("&cВы были забанены!\n\n&fПереподключитесь, чтобы узнать подробности."));

            // Информируем модератора
            sender.sendMessage(Formatters.chat("&aИгрок &f" + target + "&a забанен " + infoExpiration + " &aпо причине: &f" + reason));

            // Информируем всех игроков
            if (plugin.getConfig().getBoolean("broadcast_ban_message"))
                Bukkit.broadcastMessage(Formatters.chat("&c&lОдин из игроков был удалён из игры за использование читов или другие нарушения. " +
                        "\n&bБлагодарим вас за жалобы!"));

            return true;
        }

        return false;
    }

}

File: src/main/kotlin/net/defracted/adminutils/commands/Ban.kt
package net.defracted.adminutils.commands

import net.defracted.adminutils.Main
import net.defracted.adminutils.util.Formatters

import org.bukkit.BanList
import org.bukkit.Bukkit
import org.bukkit.command.Command
import org.bukkit.command.CommandExecutor
import org.bukkit.command.CommandSender

import java.util.*


// Нужно завершить
class Ban(private val plugin: Main) : CommandExecutor {
    override fun onCommand(sender: CommandSender, cmd: Command, label: String, args: Array<String>): Boolean {
        if (cmd.name.equals("ban", ignoreCase = true)) {
            if (args.size <= 0) {
                sender.sendMessage(Formatters.chat("&cВы не указали ник!"))
                return true
            }

            // Получаем цель бана
            val target = args[0]

            // Проверяем наличие длительности
            if (args.size < 2) {
                sender.sendMessage(Formatters.chat("&cВы не указали длительность бана."))
                return true
            }

            // Получаем срок бана (в минутах)
            val punishmentDuration: Int
            punishmentDuration = if (Formatters.isStrNum(args[1])) {
                args[1].toInt()
            } else {
                sender.sendMessage(Formatters.chat("&cУкажите срок действия бана в минутах!"))
                return true
            }

            // Создаём дату, основываясь на указанном времени
            val punishmentDateExpiration: Date?
            punishmentDateExpiration = if (punishmentDuration > 0) {
                Date(System.currentTimeMillis() + punishmentDuration * 60000)
            } else {
                null
            }

            // Получение причины из аргументов
            var reason = ""
            if (args.size <= 2) {
                reason += "не указана."
            } else {
                for (i in 2 until args.size) reason += args[i] + if (i != args.size - 1) " " else ""
            }
            val infoExpiration: String
            infoExpiration = if (punishmentDuration <= 0) {
                "&fнавсегда"
            } else {
                "&fдо " + punishmentDateExpiration.toString()
            }

            // Кикаем цель, но если игрок в онлайне
            val playerTarget = Bukkit.getPlayerExact(target)

            // Баним цель на указанный срок
            Bukkit.getBanList(BanList.Type.NAME).addBan(target, reason, punishmentDateExpiration, sender.name)

            // Кикаем игрока, если он в онлайне
            playerTarget?.kickPlayer(Formatters.chat("&cВы были забанены!\n\n&fПереподключитесь, чтобы узнать подробности."))

            // Информируем модератора
            sender.sendMessage(Formatters.chat("&aИгрок &f$target&a забанен $infoExpiration &aпо причине: &f$reason"))

            // Информируем всех игроков
            if (plugin.config.getBoolean("broadcast_ban_message")) Bukkit.broadcastMessage(Formatters.chat("""
    &c&lОдин из игроков был удалён из игры за использование читов или другие нарушения. 
    &bБлагодарим вас за жалобы!
    """.trimIndent()))
            return true
        }
        return false
    }

}